src/jit/gentree.cpp

```diff
@@ -3121,8 +3121,8 @@ bool Compiler::gtMarkAddrMode(GenTree* addr, int* pCostEx, int* pCostSz, var_typ
             }
             else
             {
-                assert(op2);
-                assert(op2->gtOper == GT_LSH || op2->gtOper == GT_MUL);
+                assert(op2 != nullptr);
```

Comment:
I'd drop this assert

Reply:
I prefer to get an assert failure rather than a crush.

Reply:
Agreed; given that this is the result of a walk, it's not transparently obvious that `op2` will be non-null, and though others seem to prefer to have separate asserts, I'd probably keep them together - it's all about asserting that the result of the walk is logical.